ruby/examples/lib/apply_order_discount.rb

```diff
@@ -0,0 +1,44 @@
+class ApplyOrderDiscount
+  Request = Struct.new(:id)
+  Response = Struct.new(:discount, :has_discount?)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
```

Comment:
If you take this code and try to run it in another system, by extracting it and putting it into a Gem file. However you don't need the Gateway over there, what will you need to do?

*Modify this class*

This violates the open/closed principle